docs/DeveloperGuide.md

```diff
@@ -1,34 +1,442 @@
+  * [Adding and Viewing a Patient's Visit Records](#adding-viewing-and-deleting-a-patients-visit-records)
+  * [Saving Data](#saving-data)
+<!-- ^ The above table of content is auto generated -->
+<!-- * [Instructions for manual testing](#instructions-for-manual-testing) -->
 
+## Introduction
+
+Patient Manager is a _Command Line Interface_ (CLI) application for **_general practitioners_** (GP)
+who work in clinics to manage their patient list. This includes a recording/retrieval of
+past record of visit, scheduling of the next appointment, and some other features listed below.
+With the Patient Manager, GPs will be able to reduce paperwork and have a more efficient way
+to organize the records of their patients.
+
+<!-- TODO: add a description on this DG -->
+
+## Setting up the project in your computer
+
+First, **fork** [this repo](https://github.com/AY2021S2-CS2113T-W09-4/tp), and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+
+1. Configure the JDK: Follow the guide 
+   [Intellij IDEA: Configuring the JDK @SE-EDU/guides](https://se-education.org/guides/tutorials/intellijJdk.html) \
+   IDEA: Configuring the JDK to ensure Intellij is configured 
+   to use JDK 11.
+1. Import the project as a Gradle project: Follow the guide 
+   [Intellij IDEA: Importing a Gradle project @SE-EDU/guides](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) \
+   IDEA: Importing a Gradle project to import 
+   the project into IDEA.
+   > ❗ Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. Verify the setup: Run `seedu.duke.PatientManager` and try a few commands.\
+   [Run the tests](#appendix-e-instructions-for-manual-testing) to ensure they all pass.
+
+## Design & Implementation
+
+### Architecture
+<img src="./images/architecture_diagram.png" width="600">
+
+The Architecture Diagram shown above gives a high-level explanation of Patient Manager.
+Given below is a brief overview of each component.
+
+`Main` contains the class `PatientManager`.
+
+This class is responsible for:
+- When the app is launched: Initializing the other components in the correct sequence and connecting them with each other
+- When the app exits: Shuts down the components and invokes cleanup methods where necessary
+
+`Commons` contains constants that are shared across the other classes.
+
+`UI` is responsible for displaying the all messages generated by Patient Manager to the screen.
+
+`Logic` parses and executes commands.
+
+`Model` contains the data of Patient Manager in memory.
+
+`Storage` reads data from, and writes data to, the hard disk.
+
+#### How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add S1234567D`.
+
+<img src="./images/sequence_diagram_overview.png" width="750">
```

Comment:
![image](https://user-images.githubusercontent.com/59783150/113377109-f4bf5980-93a5-11eb-8c48-3d14c9b256c9.png)

May want to use something more generic like: input("add < matric number >"), just a minor suggestion.

Also, there is an empty method call from :Ui to :Logic